app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php

```diff
@@ -62,6 +62,10 @@ public function execute(Quote $cart, array $cartItemData): void
             throw new GraphQlNoSuchEntityException(__('Could not find a product with SKU "%sku"', ['sku' => $sku]));
         }
 
+        if ((int) $product->getVisibility() === \Magento\Catalog\Model\Product\Visibility::VISIBILITY_NOT_VISIBLE) {
```

Comment:
Hi @pmclain,
Looks like we need use model like it was in
https://github.com/magento/graphql-ce/pull/752/files#diff-dbe98b6f5d79c40dbf020c0865de14eeR53

Thanks

Reply:
@naydav Good catch. I noticed the visibility in the linked PR would return an error is sending a product with search only visibility. The update to this PR uses [getVisibleInSiteIds](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Catalog/Model/Product/Visibility.php#L80-L83) instead and I've opened https://github.com/magento/graphql-ce/pull/760 making a similar adjustment to send friend.